src/popup/modules/InitQrCode.js

```diff
@@ -56,6 +56,14 @@ const gettingSelection = AddonSettings.get("autoGetSelectedText").then((autoGetS
     });
 });
 
+// check for clipboard text
+const gettingClipboard = AddonSettings.get("autoGetClipboardContent").then((autoGetClipboardContent) => {
+    if (autoGetClipboardContent !== true) {
+        return Promise.reject(new Error("using clipboard content is disabled"));
+    }
+    return navigator.clipboard.readText();
```

Comment:
[I previously outlined one import test case](https://github.com/rugk/offline-qr-code/issues/282#issuecomment-1199975970):
> If the clipboard is empty, it should IMHO still fallback/use the website URL then as usual!

Did you test that?
If it does not work, could you please adjust it to do that? (I imagine you could just check the return value of that API here and check if it's empty and reject the Promise if needed.)